pkg/chartutil/requirements.go

```
@@ -266,3 +268,128 @@ func ProcessRequirementsEnabled(c *chart.Chart, v *chart.Config) error {
+func processImportValues(c *chart.Chart, v *chart.Config) error {
+	reqs, err := LoadRequirements(c)
+	if err != nil {
+		log.Printf("Warning: ImportValues cannot load requirements for %s", c.Metadata.Name)
+		return nil
+	}
+	cvals, err := CoalesceValues(c, v)
+	if err != nil {
+		log.Fatalf("Error coalescing values for ImportValues %s", err)
+	}
+	nv := v.GetValues()
+	b := make(map[string]interface{})
+	for kk, v3 := range nv {
+		b[kk] = v3
+	}
+	for _, r := range reqs.Dependencies {
+		if len(r.ImportValues) > 0 {
+			var outiv []interface{}
+			for _, riv := range r.ImportValues {
+				switch riv.(type) {
+				case map[string]interface{}:
+					if m, ok := riv.(map[string]interface{}); ok {
+						nm := make(map[string]string)
+						nm["child"] = m["child"].(string)
+						nm["parent"] = m["parent"].(string)
+						outiv = append(outiv, nm)
+						s := r.Name + "." + nm["child"]
+						vv, err := cvals.Table(s)
+						if err != nil {
+							log.Printf("Warning: ImportValues missing table %v", err)
+							continue
+						}
+						if nm["parent"] == "." {
+							coalesceTables(b, vv.AsMap())
+						} else {
+
+							vm := pathToMap(nm["parent"], vv.AsMap())
+							coalesceTables(b, vm)
+						}
+					}
+				case string:
+					nm := make(map[string]string)
+					nm["child"] = "exports." + riv.(string)
+					nm["parent"] = "."
+					outiv = append(outiv, nm)
+					s := r.Name + "." + nm["child"]
+					vv, err := cvals.Table(s)
+					if err != nil {
+						log.Printf("Warning: ImportValues missing table %v", err)
+						continue
+					}
+					coalesceTables(b, vv.AsMap())
+				}
+			}
+			// set our formatted import values
+			r.ImportValues = outiv
+		}
+	}
+
+	cv, err := coalesceValues(c, b)
+	if err != nil {
+		log.Fatalf("Error coalescing processed values for ImportValues %s", err)
+	}
+	y, err := yaml.Marshal(cv)
+	if err != nil {
+		log.Printf("Warning: ImportValues could not marshall %v", err)
```

- [ ] Why not return an error here?